packages/trace-viewer/src/traceModelBackends.ts

```diff
@@ -133,7 +133,7 @@ export class FetchTraceModelBackend implements TraceModelBackend {
 }
 
 function formatUrl(trace: string) {
-  let url = trace.startsWith('http') || trace.startsWith('blob') ? trace : `file?path=${trace}`;
+  let url = trace.startsWith('http') || trace.startsWith('blob') ? trace : `file?path=${encodeURIComponent(trace)}`;
```

Comment:
Do we `decodeURIComponent` on the http server side?

Reply:
We do, but extracted into https://github.com/microsoft/playwright/pull/23893.